packages/devtools_app/lib/src/screens/inspector_v2/inspector_controller.dart

```diff
@@ -431,13 +442,29 @@ class InspectorController extends DisposableController
       inspectorTree.root = rootNode;
 
       refreshSelection(newSelection);
+      _implementationWidgetsHidden.value = hideImplementationWidgets;
     } catch (error, st) {
       _log.shout(error, error, st);
       treeGroups.cancelNext();
       return;
     }
   }
 
+  Future<void> toggleImplementationWidgetsVisibility() async {
+    final hideImplementationWidgets = !_implementationWidgetsHidden.value;
```

Comment:
Nit: consider just doing `await _recomputeTreeRoot(root, hideImplementationWidgets: !_implementationWidgetsHiddenValue.value)` or moving this line into the conditional statement.

Reply:
Done!